javascript/javascript1/week4/voiceAssistant.js

```diff
@@ -0,0 +1,133 @@
+let myName = "";
+const toDoList = [];
+
+function getReply(command) 
+{
+    let x = command.toLowerCase();
+    const start = x.indexOf("is");
+    const name = x.substring(start + 2).trim();
+if (x.includes("my name is")) 
+{
+    if (name == "") 
+    {
+        console.log('Please enter your name')
+        myName = prompt('What is your name?')
+        return `Nice to meet you ${myName}`;
+    } 
+    else {
+    myName = name
+    return `Nice to meet you ${myName}`;
+    }
+} 
+else if (x.includes("what is my name")) 
+{
+    return `Your name is ${myName}`;
+} 
+else if (x.includes("to my todo")) 
+{
+    const start = x.indexOf("add");
+    const end = x.indexOf("to my todo");
+    const toDoItem = x.substring(start + 4, end - 1);
+    toDoList.push(toDoItem);
+    return `${toDoItem} added to your todo`;
+
+} 
+else if (x.includes("from my todo"))
+{
+    const begin = x.indexOf("remove")
+    const last = x.indexOf("from my todo");
+    const toDoItem = x.substring(begin + 7, last - 1);
+    const index = toDoList.indexOf(toDoItem);
+    if (index !== -1) 
+    {
+        toDoList.splice(index, 1);
+        return `${toDoItem} removed from your todo`;
+    } 
+    else
+    {
+        return `${toDoItem} is not in your todo`;
+    }
+} 
+else if (x.includes("what is on my todo"))
+{
+    let listOfItem = "";
+    for (let i = 0; i < toDoList.length; i++) 
+    {
+        if (i === 0) {
+            listOfItem = toDoList[i];
+        } 
+        else if (i === toDoList.length - 1)
+        {
+            listOfItem = `${listOfItem}  and ${toDoList[i]}`;
+        } 
+        else 
+        {
+            listOfItem = `${listOfItem} , ${toDoList[i]}`;
+        }
+    }
+    return `you have ${toDoList.length} todo's - ${listOfItem}`;
+} 
+else if (x.includes("what day is it today"))
+{
+    const today = new Date();
+    const dateArr = today.toString().split(" ");
+    return `${dateArr[2]}. of ${dateArr[1]} ${dateArr[3]}`;
+}
+else if (x.includes("what is")) 
+{
+    const arr = x.split(" ");
+    const num1 = parseInt(arr[2]);
+    const num2 = parseInt(arr[4]);
```

Comment:
There's a trick called "array destructuring", where you assign variables directly from the array:
```js
//       "what"          "is"        "2"       "+"        "2"
const [ firstElement, secondElement, number1, operator, number2 ] = x.split(" ");
// and since you don't need the first two elements, you can assign them to nothing:
const [ , , number1, operator, number2 ] = x.split(" ");
```